usaspending_api/data_load/field_mappings_fpds.py

```diff
@@ -0,0 +1,543 @@
+    "foundation": "foundation",
+    "woman_owned_business": "woman_owned_business",
+    "minority_owned_business": "minority_owned_business",
+    "women_owned_small_business": "women_owned_small_business",
+    "economically_disadvantaged": "economically_disadvantaged",
+    "joint_venture_women_owned": "joint_venture_women_owned",
+    "joint_venture_economically": "joint_venture_economically",
+    "veteran_owned_business": "veteran_owned_business",
+    "service_disabled_veteran_o": "service_disabled_veteran_o",
+    "contracts": "contracts",
+    "grants": "grants",
+    "receives_contracts_and_gra": "receives_contracts_and_gra",
+    "airport_authority": "airport_authority",
+    "council_of_governments": "council_of_governments",
+    "housing_authorities_public": "housing_authorities_public",
+    "interstate_entity": "interstate_entity",
+    "planning_commission": "planning_commission",
+    "port_authority": "port_authority",
+    "transit_authority": "transit_authority",
+    "subchapter_s_corporation": "subchapter_s_corporation",
+    "limited_liability_corporat": "limited_liability_corporat",
+    "foreign_owned_and_located": "foreign_owned_and_located",
+    "american_indian_owned_busi": "american_indian_owned_busi",
+    "alaskan_native_owned_corpo": "alaskan_native_owned_corpo",
+    "indian_tribe_federally_rec": "indian_tribe_federally_rec",
+    "native_hawaiian_owned_busi": "native_hawaiian_owned_busi",
+    "tribally_owned_business": "tribally_owned_business",
+    "asian_pacific_american_own": "asian_pacific_american_own",
+    "black_american_owned_busin": "black_american_owned_busin",
+    "hispanic_american_owned_bu": "hispanic_american_owned_bu",
+    "native_american_owned_busi": "native_american_owned_busi",
+    "subcontinent_asian_asian_i": "subcontinent_asian_asian_i",
+    "other_minority_owned_busin": "other_minority_owned_busin",
+    "for_profit_organization": "for_profit_organization",
+    "nonprofit_organization": "nonprofit_organization",
+    "other_not_for_profit_organ": "other_not_for_profit_organ",
+    "us_local_government": "us_local_government",
+}
+
+transaction_fpds_functions = {}
+
+# broker column name -> usaspending column name
+transaction_normalized_columns = {
+    "federal_action_obligation": "federal_action_obligation",
+    "period_of_performance_star": "period_of_performance_start_date",
+    "period_of_performance_curr": "period_of_performance_current_end_date",
+    "action_date": "action_date",
+    "action_type": "action_type",
+    "action_type_description": "action_type_description",
+    "federal_action_obligation": "federal_action_obligation",
+    "award_description": "description",
+    "last_modified": "last_modified_date",
+    "award_modification_amendme": "modification_number",
+    "unique_award_key": "unique_award_key",
+}
+
+# usaspending column name -> derivation function
+transaction_normalized_functions = {
```

Comment:
I think you could make this a bit cleaner and more manageable / reasonable by consolidating the column dictionary maps into one dictionary per destination table (i.e. columns+bools+functions), and then as the right-hand-side is processed, detect whether it is a callable or not, and if so, the truthy/falsey callable. If not, don't call it. If so, call it.

The RHS might need to be a tuple of `(transform_func_to_call, str_dest_table_col_to_put_the_result_into)`.

Reply:
This still wouldn't account for the difference between the strings we treat as strings but capitalize or the strings we treat as nullable booleans but treat null as false, but I see where you're going with this. I kind of debate which one ends up looking cleaner. I'll play around with ideas if I make more process on getting this working at all